### ISSUE TITLE
JSONB returned as String returning field of a mutation iff it includes nested fields

### ISSUE THREAD

**Issue by fvieira** [hasura/graphql-engine#3682]
When doing `update`, `insert` or `delete` mutations and requesting a JSONB field inside the `returning` field, it will either return as JSON or as a String depending on whether I request some nested field or not.

Example without nested field (area is the JSONB field):

```graphql
mutation {
  update_zone(_set: {name: "Test"}, where: {id: {_eq: 21}}) {
    returning {
      area
    }
  }
}
```

```json
{
  "data": {
    "update_zone": {
      "returning": [
        {
          "area": [
            [
              -8.019762,
              39.5492578
            ],
            [
              -8.0088615,
              39.5483313
            ],
            [
              -8.005085,
              39.5497873
            ]
          ]
        }
      ]
    }
  }
}
```

Example with nested field:

```graphql
mutation {
  update_zone(_set: {name: "Test"}, where: {id: {_eq: 21}}) {
    returning {
      area
      ambassador {
        id
      }
    }
  }
}
```

```json
{
  "data": {
    "update_zone": {
      "returning": [
        {
          "area": "[[-8.019762, 39.5492578], [-8.0088615, 39.5483313], [-8.005085, 39.5497873]]",
          "ambassador": {
            "id": 1
          }
        }
      ]
    }
  }
}
```

I'm on version 1.0.0 of Hasura, and while I found a PR (https://github.com/hasura/graphql-engine/pull/3375) which seems to tackle this exact problem and that I think has been merged in this version already, I'm still seeing this behaviour...

**Comment by rikinsk** [comment]
@fvieira PR (#3375) fixes this issue as you mentioned. This fix has been merged to master but has not made it to a release yet. This will be going out in v1.1.0 which should be released quite soon.

**Comment by fvieira** [comment]
You're right, I misread the git history, the PR was merged to master before the 1.0.0 release, but the release didn't include it, sorry for the confusion.
